src/commands/githubActions/endToEndTest/executeTests.ts

```diff
@@ -0,0 +1,229 @@
+    .catch(() => {
+      // Bypass error
+    });
+
+  // Install package from scratch
+  await dappmanagerTestApi.packageInstall({
+    name: dnpName,
+    version: releaseMultiHash
+  });
+
+  await executeTestCheckers({
+    dnpName,
+    compose,
+    errorLogsTimeout,
+    healthCheckUrl
+  });
+}
+
+/**
+ * Installs the latest published version of a package
+ * then updates it with the given hash. If the package is already installed,
+ * it will be removed first.
+ */
+async function packageUpdateTest({
+  dappmanagerTestApi,
+  dnpName,
+  releaseMultiHash,
+  compose,
+  errorLogsTimeout,
+  healthCheckUrl
+}: {
+  dappmanagerTestApi: DappmanagerTestApi;
+  dnpName: string;
+  releaseMultiHash: string;
+  compose: Compose;
+  errorLogsTimeout: number;
+  healthCheckUrl?: string;
+}): Promise<void> {
+  //  Remove package if installed and bypass error if any
+  await dappmanagerTestApi
+    .packageRemove({ dnpName, deleteVolumes: true })
+    .catch(() => {
+      // Bypass error
+    });
+
+  // Install package from scratch
+  await dappmanagerTestApi.packageInstall({
+    name: dnpName
+  });
+
+  // Update package to the given hash
+  await dappmanagerTestApi.packageInstall({
+    name: dnpName,
+    version: releaseMultiHash
+  });
+
+  await executeTestCheckers({
+    dnpName,
+    compose,
+    errorLogsTimeout,
+    healthCheckUrl
+  });
+}
+
+async function executeTestCheckers({
+  dnpName,
+  compose,
+  errorLogsTimeout,
+  healthCheckUrl
+}: {
+  dnpName: string;
+  compose: Compose;
+  errorLogsTimeout: number;
+  healthCheckUrl?: string;
+}): Promise<void> {
+  const docker = new Docker();
+  for (const service of Object.keys(compose.services)) {
+    const containerName = getContainerName({
+      dnpName,
+      serviceName: service,
+      isCore: false
+    });
+    await ensureContainerStatus(containerName, docker);
+    await ensureNoErrorLogs(containerName, docker, errorLogsTimeout);
+  }
+
+  if (healthCheckUrl) await ensureHealthCheck(healthCheckUrl);
+}
+
+async function ensureContainerStatus(
+  containerName: string,
+  docker: Docker
+): Promise<void> {
+  const container = await docker.getContainer(containerName).inspect();
+  if (!container.State.Running)
+    throw Error(`Container ${containerName} is not running`);
+  console.log(chalk.green(`Container ${containerName} is running`));
+}
+
+async function ensureHealthCheck(healthCheckUrl: string): Promise<void> {
+  const res = await got(healthCheckUrl);
+  if (res.statusCode !== 200)
+    throw Error(`Healthcheck endpoint returned ${res.statusCode}`);
+  console.log(chalk.green(`Healthcheck endpoint returned 200`));
+}
+
+async function ensureNoErrorLogs(
+  containerName: string,
+  docker: Docker,
+  errorLogsTimeout: number
+): Promise<void> {
+  // maximum number of seconds to wait for the container to be running smoothly and check its logs are 120 seconds
+  errorLogsTimeout > 120 && (errorLogsTimeout = 120);
```

Comment:
You might want to have this as a constant in a params file?

Reply:
No since its only been used there